lib/stackify-api-ruby.rb

```diff
@@ -106,23 +115,36 @@ def internal_log level, msg
     def run
       Stackify::Utils.is_api_enabled
       if Stackify.configuration.api_enabled
-        if Stackify.is_valid?
-          at_exit { make_remained_job }
-           t1 = Thread.new { Stackify.authorize }
+        # check transport types
+        case Stackify.configuration.transport
+        when Stackify::DEFAULT
+          if Stackify.is_valid?
+            at_exit { make_remained_job }
+             t1 = Thread.new { Stackify.authorize }
+            case Stackify.configuration.mode
+            when MODES[:both]
+              t2 = start_logging
+              t3 = start_metrics
+            when MODES[:logging]
+              t2 = start_logging
+            when MODES[:metrics]
+              t3 = start_metrics
+            end
+
+            t1.join
+            t3.join if t3
+          else
+            Stackify.log_internal_error "Stackify is not properly configured! Errors: #{Stackify.configuration.errors}"
+          end
+        when Stackify::UNIX_SOCKET
```

Comment:
configuration for unix domain socket currently only affects the logging portion.  The library should continue to support metrics being uploaded via the api even if the unix domain socket is setup for logging